### PR TITLE
Remove superseded features ahead of v0.13 release

### LINKED ISSUE #1702: Stop generating and installing OAM CRDs
### What problem are you facing?

We are currently [generating](https://github.com/crossplane/crossplane/blob/365a7c3ad900ff6cf8da646fedb99e894e07c41e/apis/generate.go#L28) OAM CRDs and [installing](https://github.com/crossplane/crossplane/tree/master/cluster/charts/crossplane/templates/crds) them as part of the Crossplane helm chart. This will cause issues when the oam-kubernetes-runtime helm chart installs the CRDs as part of its helm chart.

### How could Crossplane help solve your problem?

Remove the OAM CRDs from core Crossplane.

### PR DESCRIPTION
**Pull request by negz**
### Description of your changes

Fixes #1702
Updates #1670

This PR removes:

* The resource claim type definitions, which were deprecated in v0.12.
* The 'autotarget' controller, which created a `KubernetesTarget` for each `KubernetesCluster` claim.
* The OAM controllers and CRDs, which will now be installed by the [oam-kubernete-runtime](https://github.com/crossplane/oam-kubernetes-runtime/) Helm chart.
* The original package manager, which has been replaced in v0.13.

The removal of the original package manager was not originally planned for v0.13, but keeping it around makes little sense given the removal of resource claims. See the relevant commit message for more context.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
I've built and deployed this branch, along with https://github.com/crossplane/provider-gcp/pull/254 (which is built against this branch and https://github.com/crossplane/crossplane-runtime/pull/201). All seems to be working as expected when I run through the GCP composition examples.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: cluster/charts/crossplane/README.md [hunk 1]
```diff
@@ -272,30 +112,12 @@ and their default values.
...
-| `clusterPackages.azure.deploy`     | Deploy Azure package                                              | `false`
-| `clusterPackages.azure.version`    | Azure provider version to deploy                                   | `<latest released version>`
-| `clusterPackages.rook.deploy`      | Deploy Rook package                                               | `false`
-| `clusterPackages.rook.version`     | Rook provider version to deploy                                    | `<latest released version>`
-| `personas.deploy`                | Install roles and bindings for Crossplane user personas         | `true`
-| `templateStacks.enabled`         | Enable experimental template stacks support                     | `true`
-| `templateStacks.controllerImage` | Template Stack controller image                                 | `crossplane/templating-controller:v0.2.1`
-| `priorityClassName      `        | Priority class name for crossplane and package manager pods       | `""`
+| `priorityClassName`        | Priority class name for crossplane and package manager pods       | `""`
```
Comment (hasheddan) [inline cluster/charts/crossplane/README.md:115]:
```suggestion
| `priorityClassName`        | Priority class name for crossplane pod       | `""`
```

File: go.mod [hunk 2]
```diff
@@ -4,19 +4,15 @@ go 1.13
 
 require (
 	github.com/alecthomas/kong v0.2.11
+	github.com/alecthomas/template v0.0.0-20190718012654-fb15b899a751 // indirect
+	github.com/alecthomas/units v0.0.0-20190924025748-f65c72e2690d // indirect
```
Comment (hasheddan) [inline go.mod:8]: Do you know what introduced these?

File: Makefile [hunk 3]
```diff
@@ -112,41 +111,6 @@ manifests.prepare:
 	cp $(TYPE_MANIFESTS) $(CROSSPLANE_CHART_DIR)/templates
 	@$(OK) Copied type manifests to Crossplane chart
 
-
-# Add "helm.sh/hook: crd-install" and "helm.sh/hook-delete-policy:
```
[the remark this answers is not in the document]
Comment (negz) [reply Makefile:116]: I'd prefer to approach that in a separate PR - I'd prefer this PR to be strictly removing functionality as it's quite large.

File: go.mod (hunk 2 above)
Comment (negz) [reply go.mod:8]: iirc they appeared when the dependency on `oam-kubernetes-runtime` was removed.

File: Makefile (hunk 3 above)
[the remark this answers is not in the document]
Comment (hasheddan) [reply Makefile:116]: SGTM

File: go.mod (hunk 2 above)
Comment (hasheddan) [reply go.mod:8]: Hmm that is somewhat surprising to me :thinking: I guess if they don't flag `make check-diff` then they are accurate though...

File: cluster/charts/crossplane-types/templates/clusterrole.yaml [hunk 4]
```diff
@@ -66,15 +62,8 @@ rules:
   - list
   - watch
 - apiGroups:
-  - cache.crossplane.io
-  - compute.crossplane.io
-  - database.crossplane.io
   - kubernetes.crossplane.io
```
Comment (muvaf) [inline cluster/charts/crossplane-types/templates/clusterrole.yaml:65]: I remember a PR removing Kubernetes Provider type usage from other repos. Do we need to keep it here?

File: cluster/charts/crossplane/README.md [hunk 5]
```diff
@@ -88,166 +88,6 @@ helm install crossplane --namespace crossplane-system crossplane-master/crosspla
 helm install crossplane --namespace crossplane-system crossplane-master/crossplane --version <version> --devel --disable-openapi-validation
 ```
 
-## Installing Infrastructure Providers
```
Comment (muvaf) [inline cluster/charts/crossplane/README.md:91]: Why are these parts removed?

File: cluster/charts/crossplane/README.md [hunk 6]
```diff
@@ -272,30 +112,12 @@ and their default values.
 | `imagePullSecrets`               | Names of image pull secrets to use                              | `dockerhub`                                            |
 | `replicas`                       | The number of replicas to run for the Crossplane operator       | `1`                                                    |
 | `deploymentStrategy`             | The deployment strategy for the Crossplane operator             | `RollingUpdate`                                        |
-| `clusterPackages.aws.deploy`       | Deploy AWS package                                                | `false`
```
Comment (muvaf) [inline cluster/charts/crossplane/README.md:275]: Do we remove the feature that allows providers to be installed together with Crossplane?

File: cluster/charts/crossplane-controllers/values.yaml.tmpl [hunk 7]
```diff
@@ -41,14 +37,6 @@ resourcesCrossplane:
...
-  limits:
-    cpu: 100m
-    memory: 512Mi
-  requests:
-    cpu: 100m
-    memory: 256Mi
-
 resourcesRBACManager:
   limits:
     cpu: 100m
```
[the remark this answers is not in the document]
Comment (muvaf) [reply cluster/charts/crossplane-controllers/values.yaml.tmpl:42]: And do we capture their functionality with some other mechanism?

File: examples/README.md [hunk 8]
```diff
@@ -1,17 +0,0 @@
-# Crossplane Examples
```
Comment (muvaf) [inline examples/README.md:1]: Can we link to `examples` folder of providers somewhere? Either in `README.md` or `examples/README.md`? I believe the shortest path to the actual examples should be as short as possible. An `examples` folder in the root of the repo is pretty good in that sense but if we remove it, at least let's make sure we have references to examples in READMEs and documentation website.

File: cluster/charts/crossplane/README.md (hunk 5 above)
Comment (hasheddan) [reply cluster/charts/crossplane/README.md:91]: @muvaf they will be changing with new package manager, I'll circle back to update them in a subsequent PR 👍

File: cluster/charts/crossplane/README.md (hunk 6 above)
Comment (hasheddan) [reply cluster/charts/crossplane/README.md:275]: We are replacing it with the new packages, tracked: https://github.com/crossplane/crossplane/issues/1726

File: examples/README.md (hunk 8 above)
Comment (hasheddan) [reply examples/README.md:1]: There are also some docs links that I think we may need to clean up as a result of this

File: cluster/charts/crossplane-types/templates/clusterrole.yaml (hunk 4 above)
Comment (negz) [reply cluster/charts/crossplane-types/templates/clusterrole.yaml:65]: I don't think so; I can remove it.

File: cluster/charts/crossplane-controllers/values.yaml.tmpl (hunk 7 above)
[the remark this answers is not in the document]
Comment (negz) [reply cluster/charts/crossplane-controllers/values.yaml.tmpl:42]: I believe they are used by the original package manager, which is removed by this commit: https://github.com/crossplane/crossplane/blob/0382319a83822d2e0b011c62d9adc018f68b9ac7/cluster/charts/crossplane-controllers/templates/package-manager-deployment.yaml

I don't think they're relevant under the new package manager design, which is opinionated about how provider controllers deployments are formed.

File: examples/README.md (hunk 8 above)
Comment (muvaf) [reply examples/README.md:1]: What do you guys think about having only `examples/README.md` that points to `examples` folder of the providers one by one?

**Comment by negz** [comment]
I’d like to merge it soon if others feel comfortable. It’s almost entirely a red diff - there are some things (examples, installing providers via the Helm chart) that I’d like us to handle as follow-on work.

File: examples/README.md (hunk 8 above)
Comment (negz) [reply examples/README.md:1]: This sounds good to me, but I'd like to keep this diff mostly red and handle that in a follow-on PR.

**Review by hasheddan — APPROVED** [review]
Thanks for taking care of this @negz! I feel good about merging then following up 👍

File: examples/README.md (hunk 8 above)
Comment (negz) [reply examples/README.md:1]: Just cleaned up any references from the docs to the examples, by moving the referenced examples (scripts, actually) into docs/snippets.

**Pull request merged**